Description: fix compilation on BigEndian
 cannot pass a function by reference where an lvalue is expected
 (only applies to bigendian, where a macro expands to a byteswap function)
Author: IOhannes m zmölnig
Bug: https://github.com/assimp/assimp/issues/613
Reviewed-by: IOhannes m zmölnig
Last-Update: 2015-08-10
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- assimp.orig/code/Bitmap.cpp
+++ assimp/code/Bitmap.cpp
@@ -84,7 +84,12 @@
 
     template<typename T>
     inline std::size_t Copy(uint8_t* data, T& field) {
+#ifdef AI_BUILD_BIG_ENDIAN
+        T field_swapped=AI_BE(field);
+        std::memcpy(data, &field_swapped, sizeof(field)); return sizeof(field);
+#else
         std::memcpy(data, &AI_BE(field), sizeof(field)); return sizeof(field);
+#endif
     }
 
     void Bitmap::WriteHeader(Header& header, IOStream* file) {
